src/huntsman/pocs/observatory.py

```diff
@@ -370,12 +370,20 @@ def _take_observation_block(self, observation, cameras, timeout=60 * u.second,
             # TODO: Replace with concurrent.futures
             events = {}
             for cam_name, camera in cameras.items():
-                events[cam_name] = camera.take_observation(observation, headers=headers,
-                                                           exptime=exptime)
+                try:
+                    events[cam_name] = camera.take_observation(observation, headers=headers,
+                                                               exptime=exptime)
+                except error.PanError as err:
+                    self.logger.error(f"{err}!r")
+                    self.logger.warning("Continuing with observation block after error.")
             # Wait for the exposures (blocking)
             # TODO: Use same timeout as camera client
-            self._wait_for_camera_events(events, duration=exptime + timeout,
-                                         remove_on_error=remove_on_error)
+            try:
+                self._wait_for_camera_events(events, duration=exptime + timeout,
+                                             remove_on_error=remove_on_error)
+            except error.Timeout as err:
+                self.logger.error(f"{err}!r")
```

Comment:
```suggestion
                self.logger.error(f"{err!r}")
```

Same for those below.